docs/source/nodes_and_pipelines/nodes.md

```diff
@@ -184,114 +184,50 @@ You can also call a node as a regular Python function: `adder_node(dict(a=2, b=3
-
-        super().__init__(
-            filepath=PurePosixPath(path),
-            version=version,
-            exists_function=self._fs.exists,
-            glob_function=self._fs.glob,
-        )
-
-        # Handle default load and save arguments
-        self._load_args = deepcopy(self.DEFAULT_LOAD_ARGS)
-        if load_args is not None:
-            self._load_args.update(load_args)
-        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
-        if save_args is not None:
-            self._save_args.update(save_args)
-
-    def _describe(self) -> Dict[str, Any]:
-        return {
-            "filepath": self._filepath,
-            "protocol": self._load_args,
-            "save_args": self._save_args,
-            "version": self._version,
-        }
-
-    def _load(self) -> pd.DataFrame:
-        load_path = str(self._get_load_path())
-        return pd.read_csv(load_path, **self._load_args)
-
-    def _save(self, data: pd.DataFrame) -> None:
-        save_path = get_filepath_str(self._get_save_path(), self._protocol)
+You can create the project with this command
+```bash
+kedro new -s pandas-iris
+```
 
-        buf = BytesIO()
-        data.to_csv(path_or_buf=buf, **self._save_args)
+### Reading data with Generators
+To utilize generator functions in Kedro nodes, you need to update the `catalog.yml` file to include the `chunksize` argument for the relevant dataset that will be processed using the generator.
 
-        with self._fs.open(save_path, mode="a+") as fs_file:
-            fs_file.write(buf.getvalue())
+You need to add a new dataset in your `catalog.yml` as follow:
+```diff
++ X_test:
++  type: pandas.CSVDataSet
++  filepath: data/05_model_input/X_test.csv
++  load_args:
++    chunksize: 10
 ```
-</details>
 
-Modify `example_iris_data` in `catalog.yml` by changing `type` to the custom dataset you created above. Add `chunksize: 100` to `load_args` which will return an iterable object. The `chunksize` parameter refers to the number of rows in each chunk.
+Thanks to `pandas` built-in support, you can use the `chunksize` argument to read data using generator.
 
-```yaml
-example_iris_data:
-  type: YOUR_PROJECT_NAME.extras.datasets.chunkwise_dataset.ChunkWiseCSVDataSet
-  filepath: data/01_raw/iris.csv
-  load_args:
-    chunksize: 100
-```
+### Saving data with Generators
+To use generator to save data lazily, you need do three things:
+- Update the `make_prediction` function definition to use `return` instead of `yield`.
+- Create a [custom dataset](../extend_kedro/custom_datasets.md) called `ChunkWiseCSVDataset`
+- Update the `catalog.yml` to use a newly created `ChunkWiseCSVDataset`.
 
-Next, in `nodes.py` we repurpose the existing `split_data` function to process chunk-wise data:
+You can copy the following code to `pipeline.py`, the main change is using a new model `DecisionTreeClassifier` to make prediction by chunks in `make_predictions`.
```

Comment:
These changes should go into `nodes.py` right and not `pipeline.py` ? 
```suggestion
You can copy the following code to `nodes.py`, the main change is using a new model `DecisionTreeClassifier` to make prediction by chunks in `make_predictions`.
```

Reply:
nice catch!